src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpConnectionFactory.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net.Connections;
+using System.Net.Sockets;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// The default connection factory used by <see cref="SocketsHttpHandler"/>, opening TCP connections.
+    /// </summary>
+    public class SocketsHttpConnectionFactory : ConnectionFactory
+    {
+        internal static SocketsHttpConnectionFactory Default { get; } = new SocketsHttpConnectionFactory();
+
+        /// <inheritdoc/>
+        public sealed override ValueTask<Connection> ConnectAsync(EndPoint? endPoint, IConnectionProperties? options = null, CancellationToken cancellationToken = default)
+        {
+            if (options == null || !options.TryGet(out DnsEndPointWithProperties? httpOptions))
+            {
+                return ValueTask.FromException<Connection>(ExceptionDispatchInfo.SetCurrentStackTrace(new HttpRequestException($"{nameof(SocketsHttpConnectionFactory)} requires a {nameof(DnsEndPointWithProperties)} property.")));
```

Comment:
SetCurrentStackTrace is really only useful when we expect the call chain and the await chain to be different, e.g. if an exception occurs, is stored into a field, and is then later thrown from that field.  Here, presumably we expect the 99% case to be `await ConnectAsync(...)`, in which case the SetCurrentStackTrace here is not only unnecessary.